core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -633,6 +633,8 @@ private void initHMC() {
           Constant.JDBC_EMBEDDED_DRIVER());
       initCommonHiveMetaStoreProperties(metadataConf);
 
+      // wait for stats sampler initialization
+      Misc.waitForSamplerInitialization();
```

Comment:
Just curious. Why is this required?

Reply:
As mentioned in other PR comments, saw some stats failures in region initialization so this ensures that stats sampler is up before opening up the catalog for users.